lib/Doctrine/DBAL/Connection.php

```diff
@@ -1597,7 +1593,7 @@ public function ping()
         }
 
         try {
-            $this->query($this->platform->getDummySelectSQL());
+            $this->query($this->getDatabasePlatform()->getDummySelectSQL());
```

Comment:
The platform isn't set anymore in `connect()` with this PR.

In the current version of `Connection` the following calls are triggered:
`getDatabasePlatform()` => `detectDatabasePlatform()` => `getDatabasePlatformVersion()`  => `connect()` at this point because the platform isn't set yet `connect()` calls `detectDatabasePlatform()` again.

This PR removes the call to `detectDatabasePlatform()` from `connect()` to stop that from happening.